storagenode/piecestore/endpoint.go

```diff
@@ -335,43 +337,43 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 	defer commitOrderToStore(ctx, &largestOrder)
 
 	// Measure upload rate and average upload rate
-	uploadSize := int64(0)
 	previousSize := int64(0)
 
 	// sumUploadRate sums all previously sampled upload rate
 	sumUploadRate := float64(0)
-	// averageUploadRate= sum of sampled upload rate / total of samples
-	averageUploadRate := float64(0)
-	// number as of data chunks that have been received so far
+
+	// chunkCount is the number as of data chunks that have been received so far
 	// first chunk is counted as 0
 	chunkCount := float64(-1)
 
+	// delayUntilSlowConnectionFlagged indicates the delay in seconds until
+	// the connnection is flagged for unusual slow upload speed.
+	// This is to avoid false positive flag in the first moment of uploading
+	delayUntilSlowConnectionFlagged := float64(10)
```

Comment:
Use a time.Duration where possible to avoid any confusion with units. For example,

    delayUntilSlowConnectionFlagged := 10 * time.Second

Reply:
fixed